### PR TITLE
Add mapfishapp addons using French BAN Geocoder API

### PR DESCRIPTION
**Pull request by pierrejego**
See #1518

### CODE REVIEW

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        // create layer : container of geometry to display adress selected
+        this.layer = new OpenLayers.Layer.Vector("__georchestra_bangeocoder", {
+            displayInLayerSwitcher: false,
+            styleMap: new OpenLayers.StyleMap({
+                "default": {
+                	graphicWidth: 20,
+                    graphicHeight: 32,
+                    graphicYOffset: -28, // shift graphic up 28 pixels
+                    //Change the location of picture if not find
+                    externalGraphic : 'app/css/images/pwrs/geoPin.png',
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js]: you should rather use 
```js
externalGraphic: GEOR.config.PATHNAME + "/app/css/images/pwrs/geoPin.png"
```
or else the path will be broken when using a permalink

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ service URL : http://api-adresse.data.gouv.fr/search/?
+ search value : "q=8 bd du port"
+ limit of responses (is set to 5 by default) : "limit=5"  
+Ex : http://api-adresse.data.gouv.fr/search/?q=8 bd du port&limit=5
+
+**** The service return GeoJSON like that :
+
+    /!\ Default SRS code is EPSG:4326 /!\
+
+{
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/README.md]: please format the js/json according to the standards (eg http://jsbeautifier.org/)

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/config.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        "enabled": true,
+        "title": {
+            "en": "Adress recentering",
+            "es": "Adress recentering",
+            "fr": "Recherche d'adresse"
+        },
+        "description": {
+            "en": "Adress recentering",
+            "es": "Adress recentering",
+            "fr": "Recherche d'adresses dans la Base d'Addresse National (France)"
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/config.json]: Nationale ;-)

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/config.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        },
+        "options": {
+            "limitResponse":5,
+            "geocodeServiceUrl":"https://api-adresse.data.gouv.fr/search/?"
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/config.json]: `geocodeServiceUrl` should be in the default options (so: in the manifest.json)

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+     * record - {Ext.data.record} a record with the addon parameters
+     */
+    init: function(record) {
+
+        // create layer : container of geometry to display adress selected
+        this.layer = new OpenLayers.Layer.Vector("__georchestra_bangeocoder", {
+            displayInLayerSwitcher: false,
+            styleMap: new OpenLayers.StyleMap({
+                "default": {
+                	graphicWidth: 20,
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js]: indents to fix

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        // get geometry
+        geom = new OpenLayers.Geometry.Point(fromCoordX, fromCoordY).transform(from, to);
+
+        // create point from geometry find in GeoJSON and create vector feature from point geometry
+        point = new OpenLayers.Geometry.Point(geom.x,geom.y);
+        feature = new OpenLayers.Feature.Vector(point);
+
+        // add point feature to layer and zoom on    
+        this.layer.addFeatures(feature);
+        this.map.setCenter(new OpenLayers.LonLat(geom.x,geom.y), 10);    },
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js]: parens position to fix (jsbeautifier to the rescue !)

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/manifest.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        "css/main.css"
+    ],
+    "default_options": {},
+    "i18n": {
+        "en": {
+            "banGeocoder.window_title": "Adresse researcher",
+            "banGeocoder.field_emptytext": "Ex : 11 Rue Louis de Broglie, 22300 Lannion",
+            "banGeocoder.field_label": "Address",
+            "Result": "Result",
+	   		"Loading...":"Loading"
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/manifest.json]: indents

**Comment by pierrejego** [comment]
Merci pour les commentaires, je fais les modifications dans la journée

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/manifest.json (hunk 7 above)
Comment (pierrejego) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/manifest.json]: Bonjour,
je pense avoir fait ce qu'il faut, il ne prend pas en compte la modification ou je me suis trompé quelque part ?

Bonne journée

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/manifest.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        "en": {
+            "banGeocoder.window_title": "Adresse researcher",
+            "Loading...":"Loading",
+            "banGeocoder.field_emptytext": "Ex : 11 Rue Louis de Broglie, 22300 Lannion",
+            "banGeocoder.field_label": "Address",
+            "Result": "Result"
+        },        
+        "es": {
+            "banGeocoder.window_title": "Adresse researcher",
+	    	"Loading...":"Loading",
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/manifest.json]: indents here and there too ...

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+     * Method: _onComboSelect
+     * Callback on combo selected to create and display address location from geometry
+     * 
+     */
+
+    _onComboSelect: function(combo, record) {
+        if (this.layer.features.length > 0) {
+            this.layer.destroyFeatures();
+        }
+        var geom, toCoordX, toCoordY, from, to, geom, fromCoordX, fromCoordY, point, feature
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js]: missing semicolon

Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js]: well, there's a problem with vars declaration here.

File: mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+     */
+    _createCbSearch: function() {
+
+        // get options from config.json file
+        var banGeocoderOptions = this.options;
+
+        // create store to pass free text and get result from service (URL)
+        var store = new Ext.data.JsonStore({
+            proxy: new Ext.data.HttpProxy({
+                url: banGeocoderOptions.geocodeServiceUrl, // set service URL in manifest.json file, more informations in README
```
Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js]: user may modify service url in config.json, not manifest.json.
manifest is written by addon maintainer.
config is under the responsability of the platform administrator

Comment (pierrejego) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js]: not sure what you want me to do, just change the comment or something else ?

Comment (fvanderbiest) [inline mapfishapp/src/main/webapp/app/addons/bangeocoder/js/main.js]: I thin you can remove it.

**Comment by pierrejego** [comment]
Ok I think everything is fixed :)  Ready for review
Let me know if there are more modifications to do.

**Comment by fvanderbiest** [comment]
Now this looks good Pierre, thanks again !

**Pull request merged**